app/views/users/new.html.erb

```diff
@@ -0,0 +1,48 @@
+<h1> Create a new user</h1>
+<p>
+    <% @user.errors.full_messages.each do |error| %>
+       <p><%= error %></p>
+    <% end %> 
+</p>
+<% if false %>
+<form method='POST' action='/users' accept-charset="UTF-8" >
+    <label>Username</label>
+    <input type='text' name='user[username]' placeholder='Your username'>
+
+    <label>Email</label>
+    <input type='email' name='user[email]' placeholder='Your email'>
+
+    <label>Password</label>
+    <input type='password' name='user[password]' >
```

Comment:
- [ ] `form_tag` gives this error:

![form_tag](https://user-images.githubusercontent.com/4246637/80923400-d9a62d00-8d83-11ea-97bc-c0e95b70dab5.png)

Using `form_tag`, there is no params[:user]. To solve it you can append the "user" in text_field_tag like so:

`text_field_tag("user[email]")`

Reply:
Maybe you should go back to the project instruction, there is a process in the project execution that explains it. You should not uncomment the commented before testing because it is part of the project at a certain state. The project is about building a form with 3 different means. The controller was also evolving as the view was changing. You can see a trace of that in the helper were we wrote a special filter for the form_tag, but the controller as it is now is meant to work with form_for, as it is the final stage of the project. So please consider reading the project instructions before you review it again.